client/state/plugins/installed/selectors.js

```diff
@@ -134,18 +200,33 @@ export function getPluginsOnSites( state, plugins ) {
 }
 
 export function getPluginOnSites( state, siteIds, pluginSlug ) {
-	return getPlugins( state, siteIds ).find( ( plugin ) => isEqualSlugOrId( pluginSlug, plugin ) );
```

Comment:
Calling `getPlugins` in all of these remaining selectors was very computationally expensive even after I reworked that selector since these are often called with different sets of siteIds thus causing the selector to recalculate numerous times, and somehow it was causing a loop where a component was rendering different sets of siteIds infinitely and thus infinitely recalculating the selector.

I changed all these selectors to use the more stable/simple data provided in `getAllPluginsIndexedBySiteId` and `getAllPluginsIndexedByPluginSlug` which greatly improve the performance of all of them and fixed the rendering loop.